bigchaindb/block.py

```diff
@@ -229,3 +232,89 @@ def _start(self):
         p_write.start()
         p_delete.start()
 
+
+class BacklogDeleteRevert(Block):
+
+    def __init__(self, q_backlog_delete):
+        # invalid transactions can stay deleted
+        self.q_tx_to_validate = q_backlog_delete
+        self.q_tx_validated = mp.Queue()
+        self.q_transaction_to_revert = mp.Queue()
+        self.q_tx_delete = mp.Queue()
+
+        self.monitor = Monitor()
+
+    def locate_transactions(self):
+        """
+        Determine if a deleted transaction has made it into a block
+        """
+        # create bigchain instance
+        b = Bigchain()
+
+        while True:
+            tx = self.q_tx_validated.get()
+
+            # poison pill
+            if tx == 'stop':
+                self.q_tx_delete.put('stop')
+                self.q_transaction_to_revert.put('stop')
+                return
+
+            # check if tx is in a (valid) block
+            validity = b.get_blocks_status_containing_tx(tx['id'])
+
+            if validity and list(validity.values()).count(Bigchain.BLOCK_VALID) == 1:
```

Comment:
Yes that is true.
I think that it would be better to instead of subclassing `Block` just duplicate the `validate_transaction` in `BacklogDeleteRevert`.

This way we don't need to add extra functionality to empty the delete queue and it becomes easier to read the code. Looking at the code it took me some time to figure out how the transactions went from `q_tx_to_validate` to `q_tx_validated`